src/app/copy-paste-node/copy-paste-node.component.ts

```diff
@@ -0,0 +1,69 @@
+import { ChangeDetectorRef, Component, HostListener, OnInit } from '@angular/core';
+import { Map, OrderedMap } from 'immutable';
+import { Subscription } from 'rxjs';
+import { clone } from 'ramda';
+import { UUID } from 'angular2-uuid';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { D3Node, UserState } from '../../non-angular/interfaces';
+import { StateService } from '../state.service';
+import { userStateServiceResponseToObject } from '../../non-angular/services-helpers';
+import { EditNodeModalComponent } from '../edit-node-modal/edit-node-modal.component';
+
+@Component({
+  selector: 'app-copy-paste-node',
+  styleUrls: ['./copy-paste-node.component.scss'],
+  templateUrl: './copy-paste-node.component.html',
+})
+export class CopyPasteNodeComponent implements OnInit {
+  node: D3Node;
+  disabled: boolean;
+  subscription: Subscription;
+  userState: UserState = {
+    copiedNodeId: '',
+    currentNode: '',
+  };
+
+  constructor(private stateService: StateService, public modalService: NgbModal, private cd: ChangeDetectorRef) {
+  }
+
+  ngOnInit() {
+    this.stateService.userState.observable.subscribe(response => {
+      this.disabled = !response.get('isEditing');
+      this.cd.markForCheck();
+      userStateServiceResponseToObject.bind(this)(response);
+      if (!this.userState.currentNode) {
+        this.userState.currentNode = '';
+      }
+    });
+  }
+
+  copyNode() {
+    this.stateService.userState.setCopiedNodeId();
+    this.subscription = this.stateService.twiglet.nodes.observable.subscribe((response: OrderedMap<string, Map<string, any>>) => {
```

Comment:
This probably needs an unsubscribe or we can just subscribe to all of the nodes in ngOnInit like we do with the userstate on line 33.